File: tree-hugger/lib/tests/fixtures/sample.go
package main

import "fmt"

// Greet greets a person by name.
// It returns a formatted greeting string.
func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// GreetMany greets multiple people.
// It takes a variadic number of names.
func GreetMany(names ...string) {
	for _, name := range names {
		fmt.Println(Greet(name))
	}
}

// Greeter is a struct that can generate greetings.
type Greeter struct {
	// Prefix is the prefix to use for greetings.
	Prefix string
}

// NewGreeter creates a new Greeter with the given prefix.
func NewGreeter(prefix string) *Greeter {
	return &Greeter{Prefix: prefix}
}

// Greet greets a person using this greeter's prefix.
func (g *Greeter) Greet(name string) string {
	return fmt.Sprintf("%s, %s!", g.Prefix, name)
}

File: tree-hugger/lib/tests/fixtures/sample.rs
/// Greets a person by name.
///
/// ## Arguments
/// * `name` - The name of the person to greet
///
/// ## Returns
/// A formatted greeting string.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

/// Greets multiple people.
///
/// ## Arguments
/// * `names` - A slice of names to greet
pub fn greet_many(names: &[String]) {
    for name in names {
        println!("{}", greet(name));
    }
}

/// A struct that can generate greetings.
pub struct Greeter {
    /// The prefix to use for greetings.
    prefix: String,
}

impl Greeter {
    /// Creates a new Greeter with the given prefix.
    pub fn new(prefix: String) -> Self {
        Self { prefix }
    }

    /// Greets a person using this greeter's prefix.
    pub fn greet(&self, name: &str) -> String {
        format!("{}, {}!", self.prefix, name)
    }
}
